lib/server-manager.ts

```diff
@@ -16,15 +16,13 @@ import { CompositeDisposable, ProjectFileEvent, TextEditor } from 'atom';
 // ChildProcess.  This is used so that language packages with alternative
 // language server process hosting strategies can return something compatible
 // with AutoLanguageClient.startServerProcess.
-export interface LanguageServerProcess extends EventEmitter {
+export interface LanguageServerProcess extends cp.ChildProcess {
```

Comment:
We definitely don't want to extend from ChildProcess here because we're trying to create an interface that exposes the minimal ChildProcess interface that we use in atom-languageclient (see the comment above the interface).  Was there some particular error you made this change to address?

More background on this: in the [ide-powershell](https://github.com/daviwil/ide-powershell) package, we use node-pty to spawn the language server process and the object it returns does not conform to the ChildProcess API shape.  I have to make a wrapper object that exposes the minimum set of methods and events that atom-languageclient uses so that the PowerShell language server can be returned from startServerProcess.  If this LanguageServerProcess interface changes to inherit from ChildProcess, it will require me to fake the whole interface even though we don't use anything else on it.

Can you think of another way to solve this issue?

Reply:
The only real issue from the type definition standpoint is that this type is eventually fed into the RPC `IPCMessageReader` and `IPCMessageWriter`. If you are sure that this will work, then we could always just perform a TypeScript cast to `ChildProcess` at those two call sites.

Reply:
Argh, yeah, that's annoying.  However it seems that their use of ChildProcess is pretty minimal as well (moreso than LanguageServerProcess) so I'd say let's cast to ChildProcess for now and then see if any future problems arise.